### PR TITLE
Starting containers from a process inside the Test's SetUp

### PR DESCRIPTION
**Pull request by WillRock19**
## What?

Running containers on StartUp of the Integration tests;

## How?

- Creating methods that run `docker-compose` process;
- Running docker-compose on SetUp of integration tests;

### CODE REVIEW

File: tests/HeroesAndVillains.Api.SuperHero.Tests.Common/Helpers/DockerComposeManager.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                    FileName = "docker-compose",
+                    Arguments = $"-f {_dockerComposeFilePath} up -d",
+                    RedirectStandardOutput = true,
+                    UseShellExecute = false,
+                    CreateNoWindow = true,
+                }
+            };
+
+            process.Start();
+            await process.WaitForExitAsync();
```
Comment (lucasteles) [inline tests/HeroesAndVillains.Api.SuperHero.Tests.Common/Helpers/DockerComposeManager.cs:69]: Por que vc escolheu sofrer ao invez de usar https://github.com/testcontainers/testcontainers-dotnet ?

Comment (lucasteles) [inline tests/HeroesAndVillains.Api.SuperHero.Tests.Common/Helpers/DockerComposeManager.cs:69]: Por que vc escolheu sofrer ao invez de usar https://github.com/testcontainers/testcontainers-dotnet ?

File: tests/HeroesAndVillains.Api.SuperHero.Tests.Integration/Configs/GlobalContext.cs [hunk 2]
```diff
@@ -1,34 +1,36 @@
...
-            EnsureContainersAreUp(containerNames);
+            var dockerComposePath = Path.Combine(Directory.GetCurrentDirectory(), "docker-compose.yml");
+
+            _dockerManager = new DockerManager(containerNames, dockerComposePath);
+            Task.Run(async () => await EnsureContainersAreUp()).Wait();
         }
 
         public void Dispose() 
         {
+            Task.Run(async () => await _dockerManager.StopAndRemoveContainers()).Wait();
```
Comment (lucasteles) [inline tests/HeroesAndVillains.Api.SuperHero.Tests.Integration/Configs/GlobalContext.cs]: Por Task run? se evita async dentro de sync, vc pode usar [IAsyncDisposable ](https://learn.microsoft.com/en-us/dotnet/api/system.iasyncdisposable?view=net-6.0) no lugar

File: tests/HeroesAndVillains.Api.SuperHero.Tests.Integration/Configs/GlobalContext.cs [hunk 3]
```diff
@@ -1,34 +1,36 @@
...
             var containerNames = new List<string>()
             {
                 "mongo-db-for-tests",
                 "azurite-for-tests"
             };
-            EnsureContainersAreUp(containerNames);
+            var dockerComposePath = Path.Combine(Directory.GetCurrentDirectory(), "docker-compose.yml");
+
+            _dockerManager = new DockerManager(containerNames, dockerComposePath);
+            Task.Run(async () => await EnsureContainersAreUp()).Wait();
```
Comment (lucasteles) [inline tests/HeroesAndVillains.Api.SuperHero.Tests.Integration/Configs/GlobalContext.cs]: `async` no construtor = 🙅🏾‍♂️🙅🏾‍♂️🙅🏾‍♂️

File: tests/HeroesAndVillains.Api.SuperHero.Tests.Integration/Configs/GlobalContext.cs (hunk 2 above)
Comment (WillRock19) [reply tests/HeroesAndVillains.Api.SuperHero.Tests.Integration/Configs/GlobalContext.cs]: Valeu a dica <3

File: tests/HeroesAndVillains.Api.SuperHero.Tests.Integration/Configs/GlobalContext.cs (hunk 3 above)
Comment (WillRock19) [reply tests/HeroesAndVillains.Api.SuperHero.Tests.Integration/Configs/GlobalContext.cs]: Usei o [IAsyncLifetime](https://www.danclarke.com/cleaner-tests-with-iasynclifetime) que tu sugeriu no Whatsapp :D

File: tests/HeroesAndVillains.Api.SuperHero.Tests.Common/Helpers/DockerComposeManager.cs (hunk 1 above)
Comment (WillRock19) [reply tests/HeroesAndVillains.Api.SuperHero.Tests.Common/Helpers/DockerComposeManager.cs:69]: Implementarei em outro PR para deixar as duas formas salvas como histórico na branch. Muito obrigado <3

**Pull request merged**